### PR TITLE
Add unit tests for `migrations/api.py` module

### PR DESCRIPTION
**Pull request by mihaelabalutoiu**
This PR adds test cases for `coriolis/migrations/api.py` module

### CODE REVIEW

File: coriolis/tests/migrations/test_api.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        storage_mappings = mock.sentinel.storage_mappings
+        replication_count = mock.sentinel.replication_count
+        shutdown_instances = mock.sentinel.shutdown_instances
+
+        result = self.api.migrate_instances(
+            self.ctxt, origin_endpoint_id, destination_endpoint_id,
+            origin_minion_pool_id, destination_minion_pool_id,
+            instance_osmorphing_minion_pool_mappings, source_environment,
+            destination_environment, instances, network_map, storage_mappings,
+            replication_count, shutdown_instances, notes=None,
```
Comment (ionutbalutoiu) [inline coriolis/tests/migrations/test_api.py:41]: Do not explicitly give the `kwargs` (`notes=None, skip_os_morphing=False, user_scripts=None`) when calling the tested function.

This way, we can cover with the unit test if those default values change, the assertion below will fail.

File: coriolis/tests/migrations/test_api.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            notes=None, skip_os_morphing=False, user_scripts=None)
+        self.assertEqual(result,
+                         self.rpc_client.migrate_instances.return_value)
+
+    def test_deploy_replica_instances(self):
+        replica_id = mock.sentinel.replica_id
+        instance_osmorphing_minion_pool_mappings = (
+            mock.sentinel.instance_osmorphing_minion_pool_mappings)
+
+        result = self.api.deploy_replica_instances(
```
Comment (ionutbalutoiu) [inline coriolis/tests/migrations/test_api.py:59]: Same as above, remove the explicit `kwargs`, and keep the same assertion.

**Review by ionutbalutoiu — CHANGES_REQUESTED** [review]
Two small comments. The rest LGTM.

File: coriolis/tests/migrations/test_api.py (hunk 1 above)
Comment (mihaelabalutoiu) [reply coriolis/tests/migrations/test_api.py]: Gotcha! You are right!

**Review by ionutbalutoiu — APPROVED** [review]

**Pull request merged**